data_engineering.qmd

```diff
@@ -1,121 +1,526 @@
+
+Moreover, in machine learning, especially in deep learning, having a diverse dataset is crucial. Synthetic data can augment existing datasets by introducing variations, thereby enhancing the robustness of models. For example, SpecAugment is an excellent data augmentation technique for Automatic Speech Recognition (ASR) systems.
+
+Pivacy and confidentiality is also a big issue. Datasets containing sensitive or personal information pose privacy concerns when shared or used. Synthetic data, being artificially generated, doesn’t have these direct ties to real individuals, allowing for safer use while preserving essential statistical properties.
+
+Generating synthetic data, especially once the generation mechanisms have been established, can be a more cost-effective alternative. In the aforementioned security application scenario, synthetic data eliminates the need for breaking multiple windows to gather relevant data.
+
+Many embedded use-cases deal with unique situations, such as manufacturing plants, that are difficult to simulate. Synthetic data allows researchers complete control over the data generation process, enabling the creation of specific scenarios or conditions that are challenging to capture in real life.
+
+While synthetic data offers numerous advantages, it is essential to use it judiciously. Care must be taken to ensure that the generated data accurately represents the underlying real-world distributions and does not introduce unintended biases.
 
-- Data Sources: crowdsourcing, pre-existing datasets etc.
-- Data Types: Structured, Semi-Structured, and Unstructured
-- Real-time Data Processing in Embedded Systems
 
 ## Data Storage
+Data sourcing and data storage go hand-in-hand and it is necessary to store data in a format that facilitates easy access and processing. Depending on the use case, there are various kinds of data storage systems that can be used to store your datasets.
+
+  ----------------------------------------------------------------------------
+                 **Database**        **Data Warehouse**    **Data Lake**
+  -------------- ------------------- --------------------- -------------------
+  **Purpose**    Operational and     Analytical            Analytical
+                 transactional                             
+
+  **Data type**  Structured          Structured            Structured,
+                                                           semi-structured
+                                                           and/or unstructured
+
+  **Scale**      Small to large      Large volumes of      Large volumes of
+                 volumes of data     integrated data       diverse data
+
+  **Examples**   MySQL               Google BigQuery,      Google Cloud
+                                     Amazon Redshift,      Storage, AWS S3,
+                                     Microsoft Azure       Azure Data Lake
```

Comment:
I feel like it might be worth touching on Vector stores/databases and their advantages over standard databases. With LLMs generating more embeddings is increasingly useful and with them so are vector databases.

https://www.pinecone.io/learn/vector-database/

Pinecone has a great summary on them